### PR TITLE
Support fuzzy search

### LINKED ISSUE #49: Better fuzzy search
In the interactive mode, entering e.g. "crome" (instead of "chrome") will show "No results...". It would be great to improve the UI to be better at fuzzy search, e.g. with the help of https://github.com/farzher/fuzzysort

### PR DESCRIPTION
**Pull request by jopemachine**
Fixes #49.

* I applied fuzzy search to `verbose`, `name` option both.
* I added a unit test for testing fuzzy search, I think it probably works in almost all cases, but not sure there is always no other such named process. In this case, the `noopProcess` could not be killed. Should I think about a better test method, or is it enough?

### CODE REVIEW

File: interactive.js [hunk 1]
```diff
@@ -99,22 +90,20 @@ const preferHeurisicallyInterestingProcesses = (a, b) => {
...
-		verbose: process_ => input ? (isWindows ? process_.name : process_.cmd).toLowerCase().includes(input.toLowerCase()) : true,
-	};
-
 	const memoryThreshold = flags.verbose ? 0 : 1;
 	const cpuThreshold = flags.verbose ? 0 : 3;
+	const fuzzySearchOption = {caseSensitive: false};
+
+	const filteredProcesses = flags.verbose
+		? new FuzzySearch(processes, [isWindows ? 'name' : 'cmd'], fuzzySearchOption).search(input)
+		: new FuzzySearch(processes, ['name'], fuzzySearchOption).search(input);
```
Comment (sindresorhus) [inline interactive.js:99]: The ternary has a lot of duplicate code. I think you can narrow the scope of the ternary.

**Comment by sindresorhus** [comment]
I think the readme should mention that it support fuzzy search.

**Comment by sindresorhus** [comment]
Tests seem to be failing.

File: interactive.js [hunk 2]
```diff
@@ -99,22 +90,24 @@ const preferHeurisicallyInterestingProcesses = (a, b) => {
...
-		verbose: process_ => input ? (isWindows ? process_.name : process_.cmd).toLowerCase().includes(input.toLowerCase()) : true,
-	};
-
 	const memoryThreshold = flags.verbose ? 0 : 1;
 	const cpuThreshold = flags.verbose ? 0 : 3;
 
-	return processes
+	const filteredProcesses = new FuzzySearch(
+		processes,
+		flags.verbose ? [isWindows ? 'name' : 'cmd'] : ['name'],
```
Comment (sindresorhus) [inline interactive.js:98]: This could be further simplified

**Comment by sindresorhus** [comment]
Tests are also failing: https://github.com/sindresorhus/fkill-cli/runs/5124963299?check_suite_focus=true

**Comment by jopemachine** [comment]
> Tests are also failing: https://github.com/sindresorhus/fkill-cli/runs/5124963299?check_suite_focus=true

I think these test failures could be related to [this Node issue](https://github.com/nodejs/node/issues/35503).
I tried to test on Windows manually but failed because modifying `process.title` doesn't work.
By any chance, would it be acceptable to test fuzzy-search in only macOS until the issue is fixed?

File: interactive.js [hunk 3]
```diff
@@ -99,22 +90,24 @@ const preferHeurisicallyInterestingProcesses = (a, b) => {
...
-		verbose: process_ => input ? (isWindows ? process_.name : process_.cmd).toLowerCase().includes(input.toLowerCase()) : true,
-	};
-
 	const memoryThreshold = flags.verbose ? 0 : 1;
 	const cpuThreshold = flags.verbose ? 0 : 3;
 
-	return processes
+	const filteredProcesses = new FuzzySearch(
+		processes,
+		flags.verbose && !isWindows ? ['cmd'] : ['name'],
```
Comment (sindresorhus) [inline interactive.js:98]: Can be further simplified:

```suggestion
		[flags.verbose && !isWindows ? 'cmd' : 'name'],
```

**Comment by sindresorhus** [comment]
> By any chance, would it be acceptable to test fuzzy-search in only macOS until the issue is fixed?

👍

**Comment by sindresorhus** [comment]
Make sure you run `$ npm test` locally before pushing commits ;)

**Pull request merged**